src/CDSAPI.jl

```diff
@@ -4,6 +4,80 @@ using HTTP
 using JSON
 using Dates
 
+using ScopedValues
+
+const KEY = ScopedValue("")
+const URL = ScopedValue("https://cds.climate.copernicus.eu/api")
+
+"""
+    credentials()
+
+Attempt to find CDS credentials using different methods:
+
+    1. direct credentials provided via a specific file
+    2. environmental variables `CDSAPI_URL` and `CDSAPI_KEY`
+    3. credential file in home directory `~/.cdsapirc`
+
+A credential file is a text file with two lines:
+
+url: https://yourendpoint
+key: your-personal-api-token
+"""
+function credentials()
+
+    dotrc = joinpath(homedir(), ".cdsapirc")
+    if isfile(dotrc)
+        url, key = credentialsfromfile(dotrc)
+    else
+        url = key = ""
+    end
+
+    # overwrite with environmental variables
+    url = get(ENV, "CDSAPI_URL", url)
+    key = get(ENV, "CDSAPI_KEY", key)
+
+    # overwrite with ScopedValues provided by user
+
+    url = isempty(URL[]) ? url : URL[]
+    key = isempty(KEY[]) ? key : KEY[]
+
+
+    if isempty(url) || isempty(key)
+        error("""
+        Missing credentials. Either add the CDSAPI_URL and CDSAPI_KEY env variables
+        or create a .cdsapirc file (default location: '$(homedir())').
+        """)
+    end
+
+    return url, key
+end
+
+"""
+    credentials(file)
+
+Parse the cds credentials from a provided file
+"""
+function credentialsfromfile(file)
+    creds = Dict()
+    open(realpath(file)) do f
+        for line in readlines(f)
+            key, val = strip.(split(line, ':', limit=2))
+            creds[key] = val
+        end
+    end
+
+    if !(haskey(creds, "url") || haskey(creds, "key")) # we can allow files with only one of the keys.
```

Comment:
Another undocumented behavior where files follow a different format than the format specified in the README. I would just enforce the presence of both URL and KEY.